.github/cloud-samples-tools/cmd/main.go

```diff
@@ -76,23 +86,58 @@ func affectedCmd(configFile string, diffsFile string) {
 	diffs := strings.Split(strings.TrimSpace(string(diffsBytes)), "\n")
 
 	// Log to stderr since GitHub Actions expects the output on stdout.
-	packages, err := config.Affected(os.Stderr, diffs)
+	paths, err := config.Affected(os.Stderr, diffs)
 	if err != nil {
 		log.Fatalln("❌ error finding the affected packages.\n", err)
 	}
-	if len(packages) > 256 {
+	if len(paths) > 256 {
 		log.Fatalln(
 			"❌ Error: GitHub Actions only supports up to 256 packages, got ",
-			len(packages),
+			len(paths),
 			" packages, for more details see:\n",
 			"https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow",
 		)
 	}
 
-	packagesJson, err := json.Marshal(packages)
+	if pathsFile != "" {
+		file, err := os.Create(pathsFile)
+		if err != nil {
+			log.Fatalln("❌ eror creating output file.\n", err)
+		}
+		for _, path := range paths {
+			fmt.Fprintf(file, "%v\n", path)
+		}
+	}
+
+	output, err := json.Marshal(paths)
 	if err != nil {
-		log.Fatalln("❌ error marshaling packages to JSON.\n", err)
+		log.Fatalln("❌ error marshaling paths to JSON.\n", err)
 	}
+	fmt.Println(string(output))
+}
 
-	fmt.Println(string(packagesJson))
+// setups command entry point to validate inputs.
```

Comment:
Same as previous: first string in comment should be the exact function name, `setupsCmd`.